WMF Framework/EventPlatformClientLibrary.swift

```diff
@@ -389,63 +386,71 @@ public class EPC: NSObject {
      * ```
      */
     private func loadStreamConfiguration(_ data: Data) {
-        guard let json = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: [String: [String: Any]]] else {
-                DDLogWarn("EPC: Problem processing JSON payload from response")
-                return
-        }
         #if DEBUG
         if let raw = String.init(data: data, encoding: String.Encoding.utf8) {
             DDLogDebug("EPC: Downloaded stream configs (raw): \(raw)")
         }
         #endif
-
-        guard let streamConfigs = json["streams"] else {
-            DDLogWarn("EPC: Problem extracting stream configs")
-            return
+        struct StreamConfigurationsJSON: Codable {
+            let streams: [String: StreamConfiguration]
         }
-
-        #if DEBUG
         do {
-            let jsonString = try streamConfigs.toPrettyPrintJSONString()
-            DDLogDebug("EPC: Processed stream configurations:\n\(jsonString)")
-        } catch let error {
-            DDLogError("EPC: \(error.localizedDescription)")
-        }
-        #endif
+            let json = try JSONDecoder().decode(StreamConfigurationsJSON.self, from: data)
 
-        // Make them available to any newly logged events before flushing
-        // buffer (this is set using serial queue but asynchronously)
-        self.streamConfigurations = streamConfigs
-
-        // Process event buffer after making stream configs available
-        // NOTE: If any event is re-submitted while streamConfigurations
-        // is still being set (asynchronously), they will just go back to
-        // input buffer.
-        var event: (stream: String, data: [String: NSCoding], domain: String?)?
-        while !self.inputBufferIsEmpty() {
-            event = self.inputBufferPopFirst()
-            if let event = event {
-                self.submit(stream: event.stream, data: event.data, domain: event.domain)
+            // Make them available to any newly logged events before flushing
+            // buffer (this is set using serial queue but asynchronously)
+            streamConfigurations = json.streams.reduce(into: [:], { (result, kv) in
+                guard let stream = Stream(rawValue: kv.key) else {
+                    return
+                }
+                result?[stream] = kv.value
+            })
+
+            // Process event buffer after making stream configs available
+            // NOTE: If any event is re-submitted while streamConfigurations
+            // is still being set (asynchronously), they will just go back to
+            // input buffer.
+            while let event = inputBufferPopFirst() {
+                guard inSample(stream: event.stream) else {
+                    continue
+                }
+                appendPostToOutputBuffer((url: streamIntakeServiceURI, body: data))
```

Comment:
`body` should be `event.data`, not `data` which is the stream configuration data.